pkgs/servers/keycloak/default.nix

```diff
@@ -6,12 +6,14 @@
 , writeText
 , nixosTests
 , callPackage
-
 , confFile ? null
 , plugins ? [ ]
+, features ? [ ]
 }:
 
-stdenv.mkDerivation rec {
+let
+  featureSubcommand = if builtins.length features > 0 then "--features=" + (lib.concatStringsSep "," features) else "";
```

Comment:
How about something along the lines of the following? :)

(though some don't like the chained `${}` I've used)

```patch
diff --git a/pkgs/servers/keycloak/default.nix b/pkgs/servers/keycloak/default.nix
index 00eb7cf2765..7df8461fed3 100644
--- a/pkgs/servers/keycloak/default.nix
+++ b/pkgs/servers/keycloak/default.nix
@@ -3,7 +3,6 @@
 , fetchzip
 , makeWrapper
 , jre
-, writeText
 , nixosTests
 , callPackage
 , confFile ? null
@@ -11,9 +10,7 @@
 , features ? [ ]
 }:
 
-let
-  featureSubcommand = if builtins.length features > 0 then "--features=" + (lib.concatStringsSep "," features) else "";
-in stdenv.mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "keycloak";
   version = "21.1.1";
 
@@ -47,7 +44,7 @@ in stdenv.mkDerivation rec {
     patchShebangs bin/kc.sh
     export KC_HOME_DIR=$(pwd)
     export KC_CONF_DIR=$(pwd)/conf
-    bin/kc.sh build ${featureSubcommand}
+    bin/kc.sh build ${lib.optionalString (features != [ ]) "--features=${lib.concatStringsSep "," features}"}
 
     runHook postBuild
   '';
```

Not blocking.